pkg/rid/store/cockroach/store.go

```diff
@@ -25,23 +25,44 @@ var (
 	DefaultTimeout = 10 * time.Second
 )
 
-// Store is an implementation of dss.Store using
-// Cockroach DB as its backend store.
+type repo struct {
+	*isaRepo
+	*subscriptionRepo
+}
+
+// Store is an implementation of store.Store using Cockroach DB as its backend
+// store.
+//
 // TODO: Add the SCD interfaces here, and collapse this store with the
 // outer pkg/cockroach
 type Store struct {
-	*ISAStore
-	*SubscriptionStore
+	db     *cockroach.DB
+	logger *zap.Logger
+	clock  clockwork.Clock
+}
 
-	db *cockroach.DB
+// Interact implements store.Interactor interface.
+func (s *Store) Interact(ctx context.Context) (repos.Repository, error) {
+	logger := logging.WithValuesFromContext(ctx, s.logger)
+
+	return &repo{
+		isaRepo: &isaRepo{
+			Queryable: s.db,
+			logger:    logger,
+		},
+		subscriptionRepo: &subscriptionRepo{
+			Queryable: s.db,
+			logger:    logger,
+			clock:     s.clock,
+		},
+	}, nil
 }
 
-// InTxnRetrier supplies a new repo, that will perform all of the DB accesses
+// Transact supplies a new repo, that will perform all of the DB accesses
 // in a Txn, and will retry any Txn's that fail due to retry-able errors
 // (typically contention).
-// Note: Currently the Newly supplied Repo *does not* support nested calls
-// to InTxnRetrier.
-func (s *Store) InTxnRetrier(ctx context.Context, f func(repo repos.Repository) error) error {
+func (s *Store) Transact(ctx context.Context, f func(repo repos.Repository) error) error {
```

Comment:
I believe we want this signature to be
```go
func (s *Store) Transact(ctx context.Context, f func(ctx context.Context, repo repos.Repository) error) error {
```
...because the transactor may want to adjust the context used by each transaction attempt.

Reply:
Yup, that makes sense. I would propose to postpone until after this PR is merged.